lib/hashie/mash.rb

```diff
@@ -359,7 +381,13 @@ def log_built_in_message(method_key)
     end
 
     def log_collision?(method_key)
-      respond_to?(method_key) && !self.class.disable_warnings? &&
+      warning_disabled = if self.class.disable_warnings_blacklist.any?
```

Comment:
Given that this block is repeated twice, I feel like we need to change the implementation of `disable_warnings?` to take a method name. That would make this method able to be:

```ruby
respond_to?(method_key) && self.class.disable_warnings?(method_key) &&
  !(regular_key?(method_key) || regular_key?(method_key.to_s))
```

What do you think?